diesel/src/pg/metadata_lookup.rs

```diff
@@ -0,0 +1,32 @@
+use std::mem;
+
+use prelude::*;
+use super::{PgConnection, PgTypeMetadata};
+
+#[allow(missing_debug_implementations)]
+pub struct PgMetadataLookup {
+    conn: PgConnection,
+}
+
+impl PgMetadataLookup {
+    pub(crate) fn new(conn: &PgConnection) -> &Self {
+        unsafe { mem::transmute(conn) }
```

Comment:
Generally a good practice, which I try to follow as well. `unsafe` makes me question these practices, though :) I guess it's okay in this case, as its intend is pretty clear and the implementation obvious ("oh hey compiler, this PgConnection reference is now a PgMetadataLookup reference"). I was more surprised to see a `new` method that returns a reference, though!